src/environment.rs

```diff
@@ -157,14 +245,49 @@ impl TestEnvironment {
 
             instruction_counter: INSTRUCTION_COUNTER_INIT,
             instruction_ids_by_label: HashMap::new(),
-        }
+        };
+
+        test_environment
+    }
+
+    /// Compiles and Publishes Packages
+    ///
+    /// IMPORTANT: Prefer usage of TestEnvironment::new(packages_map) over
+    /// TestEnvironment::new(empty_package_map) + test_environment.compile_and_publish_packages,
+    /// since the first results in caching of clean environment states + respective packages,
+    /// speeding up future calls
+    pub fn compile_and_publish_packages(&mut self, packages: HashMap<&str, PathBuf>) {
+        // Todo this should publish in a constant order. Use BTreeSet?
```

Comment:
Having proper ordering would actually be an argument to use a `vec` over a `hashmap`. Alternative would be an `IndexMap` but is another dependency which is not worth it imo.

However, I wouldn't say the different ordering is a real problem.

Reply:
We can delete that comment. That was an optimization I thought about but it's not really worth the trouble right now